src/nvim/ex_session.c

```diff
@@ -614,7 +632,10 @@ static int makeopens(FILE *fd, char_u *dirnow)
     // Similar to ses_win_rec() below, populate the tab pages first so
     // later local options won't be copied to the new tabs.
     FOR_ALL_TABS(tp) {
-      if (tp->tp_next != NULL && put_line(fd, "tabnew") == FAIL) {
+      // Use `bufhidden=wipe` to remove empty "placeholder" buffers once
+      // they are not needed. This prevents creating extra buffers (see
+      // cause of Vim patch 8.1.0829)
+      if (tp->tp_next != NULL && put_line(fd, "tabnew +setlocal\\ bufhidden=wipe") == FAIL) {
```

Comment:
Would `:tab split` would also work?

Reply:
Well, it seems like it actually does! Don't know why I didn't think about this way in the first place.

Do you want me to use `tab split` here? This will differ from already merged Vim's implementation.

Reply:
> Do you want me to use tab split here?

This PR looks fine to me. It was just an observation. :+1:
You can consider upstreaming that change if you'd like to. If it's accepted it can be ported here later.

> it creates independent extra "placholder" buffers for every tab, which might be a good idea

Not sure if that's useful. The restored buffers hide whatever buffer was being viewed in that tabpage anyway.